ScienceDirect.js

```diff
@@ -117,6 +108,46 @@ function getPDFLink(doc, onDone) {
 		}
 	}
 	
+	// Attempting to use json data, which might be available to construct a direct pdf link.
+	// A json data element might be available when using partnered institution network access.
+	// The same pdf url is created that would be available from the embedded pdf object,
+	// however the embedded object code may not always be available for parsing.
+	var jsondata = doc.querySelector('script[type="application/json"]').innerHTML;
+	if (jsondata) {
+		try {
+			jsondata = JSON.parse(jsondata);
+			Zotero.debug("Found json data");
+		}
+		catch {
+			Zotero.debug(e, 2);
+		}
+		var path = jsondata.article.pdfDownload.urlMetadata.path;
+		var pdfExtension = jsondata.article.pdfDownload.urlMetadata.pdfExtension;
+		var pii = jsondata.article.pdfDownload.urlMetadata.pii;
+		var md5 = jsondata.article.pdfDownload.urlMetadata.queryParams.md5;
+		var pid = jsondata.article.pdfDownload.urlMetadata.queryParams.pid;
+		if (path && pdfExtension && pii && md5 && pid){
+			pdfURL = doc.location.protocol + '//' + doc.location.hostname + '/' + path + '/' + pii + pdfExtension + '?md5=' + md5 + '&pid=' + pid + '&isDTMRedir=Y';
+			Zotero.debug("Created pdfURL from json data: " + pdfURL);
+			onDone(pdfURL);
+			return;
+		}
+		else {
+			Zotero.debug("Missing elements in json data required for url creation.");
+		}
+	}
+
+	// The intermediate page provides a download button that uses a canonical link to download the pdf.
+	// In case, the intermediate page fails to load, the pdf can still be downloaded directly without loading
+	// the intermediate page. By creating the same canonical url link format, the pdf can be retrieved directly. 
```

Comment:
- Capitalize acronyms (PDF, URL)
- `In case, the intermediate` → `In case the intermediate`

But I'm not following this comment. What does this have to do with the intermediate page? This block is only relevant if we didn't actually generate a URL via any method above. This isn't something that's relevant if "the intermediate page fails to load".